src/fmpz_mod_mpoly/test/t-compose_fmpz_mod_mpoly.c

```diff
@@ -0,0 +1,381 @@
+/*
+    Copyright (C) 2018 Daniel Schultz
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "test_helpers.h"
+#include "fmpz_mod_mpoly.h"
+
+TEST_FUNCTION_START(fmpz_mod_mpoly_compose, state)
+{
+    slong i, j, v;
+
+    {
+        fmpz_t p;
+        fmpz_mod_mpoly_t A, A1, A2, B;
+        fmpz_mod_mpoly_struct * Cp[3];
+        fmpz_mod_mpoly_struct C[3];
+        fmpz_mod_mpoly_ctx_t ctxAC, ctxB;
+
+        fmpz_init_set_ui(p, 100003);
+        fmpz_mod_mpoly_ctx_init(ctxB, 3, ORD_LEX, p);
+        fmpz_mod_mpoly_ctx_init(ctxAC, 2, ORD_LEX, p);
+
+        fmpz_mod_mpoly_init(B, ctxB);
+        fmpz_mod_mpoly_init(A, ctxAC);
+        fmpz_mod_mpoly_init(A1, ctxAC);
+        fmpz_mod_mpoly_init(A2, ctxAC);
+        for (i = 0; i < 3; i++)
+        {
+            Cp[i] = C + i;
+            fmpz_mod_mpoly_init(C + i, ctxAC);
+        }
+
+        fmpz_mod_mpoly_set_str_pretty(B,
+                "1 + x1*x2^2 + x2^9999999999999999999999999*x3^9", NULL, ctxB);
+
+        fmpz_mod_mpoly_set_str_pretty(C + 0, "x1 + x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 1, "x1 - x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 2, "1", NULL, ctxAC);
+        if (fmpz_mod_mpoly_compose_fmpz_mod_mpoly(A, B, Cp, ctxB, ctxAC) ||
+            fmpz_mod_mpoly_compose_fmpz_mod_mpoly_horner(A1, B, Cp, ctxB, ctxAC) ||
+            fmpz_mod_mpoly_compose_fmpz_mod_mpoly_geobucket(A2, B, Cp, ctxB, ctxAC))
+        {
+            printf("FAIL\n");
+            flint_printf("Check non-example 1\n", i);
+            fflush(stdout);
+            flint_abort();
+        }
+
+        /* fmpz_mod_mpoly_set_str_pretty(C + 0, "x1", NULL, ctxAC); */
+        /* fmpz_mod_mpoly_set_str_pretty(C + 1, "2*x2", NULL, ctxAC); */
+        /* fmpz_mod_mpoly_set_str_pretty(C + 2, "1", NULL, ctxAC); */
+        /* if (fmpz_mod_mpoly_compose_fmpz_mod_mpoly(A, B, Cp, ctxB, ctxAC) || */
+        /*     fmpz_mod_mpoly_compose_fmpz_mod_mpoly_horner(A1, B, Cp, ctxB, ctxAC) || */
+        /*     fmpz_mod_mpoly_compose_fmpz_mod_mpoly_geobucket(A2, B, Cp, ctxB, ctxAC)) */
+        /* { */
+        /*     printf("FAIL\n"); */
+        /*     flint_printf("Check non-example 2\n", i); */
+        /*     fflush(stdout); */
+        /*     flint_abort(); */
+        /* } */
```

Comment:
Here is the test case in question.

Reply:
I believe this is now resolved. After double checking the `src/nmod_mpoly/test/t-compose_nmod_mpoly.c` file I noticed the nmod tests were for the negation, I've adjusted it to mirror that. The smaller prime is not relevant to the fix. 
The test itself was at fault.